tests/pyspark-notebook/test_spark.py

```diff
@@ -3,12 +3,20 @@
 import logging
 
 from tests.conftest import TrackedContainer
-from tests.run_command import run_command
 
 LOGGER = logging.getLogger(__name__)
 
 
 def test_spark_shell(container: TrackedContainer) -> None:
     """Checking if Spark (spark-shell) is running properly"""
-    logs = run_command(container, 'spark-shell <<< "1+1"', timeout=60)
+    logs = container.run_and_wait(
+        timeout=60,
+        no_warnings=False,
+        tty=True,
+        command=["bash", "-c", 'spark-shell <<< "1+1"'],
+    )
+    warnings = TrackedContainer.get_warnings(logs)
+    assert len(warnings) == 1
+    assert "Using incubator modules: jdk.incubator.vector" in warnings[0]
```

Comment:
I guess this might disappear when we switch to JDK 21, but that's a separate story and won't be a part of the switch to Python 3.12